components/job-orchestration/job_orchestration/search_scheduler/search_scheduler.py

```diff
@@ -0,0 +1,306 @@
+        need_and = True
+    
+    query += " ORDER BY end_timestamp DESC"
+
+    cursor = db_conn.cursor()
+    cursor.execute(
+        query
+    )
+    archives_for_search = [archive_id for archive_id, in cursor.fetchall()]
+    db_conn.commit()
+    cursor.close()
+    return archives_for_search
+
+
+def get_search_tasks_for_job(
+    archives_for_search: List[tuple],
+    job_id: str,
+    search_config: SearchConfig,
+    results_cache_uri: str,
+):
+    search_config_obj = search_config.dict()
+    return group(
+        search.s(
+            job_id=job_id,
+            archive_id=archive_id,
+            search_config_obj=search_config_obj,
+            results_cache_uri=results_cache_uri,
+        ) for archive_id in archives_for_search
+    )
+
+
+def dispatch_search_job(
+    archives_for_search: List[tuple],
+    job_id: str,
+    search_config: SearchConfig,
+    results_cache_uri: str
+) -> None:
+    global active_jobs
+    task_group = get_search_tasks_for_job(archives_for_search, job_id, search_config, results_cache_uri)
+    active_jobs[job_id] = SearchJob(task_group.apply_async())
+
+
+def poll_and_submit_pending_search_jobs(db_conn, results_cache_uri: str) -> None:
+    global active_jobs
+
+    cursor = db_conn.cursor()
+    new_jobs = fetch_new_search_jobs(cursor)
+    db_conn.commit()
+    cursor.close()
+
+    for job_id, job_status, search_config, submission_time in new_jobs:
+        logger.debug(f"Got job {job_id} with status {job_status}.")
+        search_config_obj = SearchConfig.parse_obj(msgpack.unpackb(search_config))
+        archives_for_search = get_archives_for_search(db_conn, search_config_obj)
+        if len(archives_for_search) == 0:
+            if set_job_status(db_conn, job_id, JobStatus.SUCCESS, JobStatus.PENDING):
+                logger.info(f"No matching archives, skipping job {job_id}.")
+            else:
+                logger.info(f"Marking job {job_id} with no archives to search cancelled.")
+                set_job_status(db_conn, job_id, JobStatus.CANCELLED)
+            continue
+
+        dispatch_search_job(archives_for_search, str(job_id), search_config_obj, results_cache_uri)
+        if set_job_status(db_conn, job_id, JobStatus.RUNNING, job_status):
+            logger.info(f"Dispatched job {job_id} with {len(archives_for_search)} archives to search.")
+        else:
+            logger.error(f"Marking job {job_id} with {len(archives_for_search)} archives to search cancelled.")
+            cancel_job(job_id)
+            set_job_status(db_conn, job_id, JobStatus.CANCELLED)
+
+
+def try_getting_results(result):
+    '''
+    https://github.com/celery/celery/issues/4084
+    Ideally we'd just do it this way, but because of the open bug listed above we have to use this
+    timeout based approach until we switch to redis result backend.
+    if not result.ready():
+        return None
+    return result.get()
+    '''
```

Comment:
```suggestion
"""
    Ideally, we'd use this code:
    
    if not result.ready():
        return None
    return result.get()
    
    But because of https://github.com/celery/celery/issues/4084, we have to use the following
    timeout based approach until we switch to the Redis result backend.
    """
```